arangod/Indexes/Index.cpp

```diff
@@ -261,6 +261,20 @@ size_t Index::sortWeight(arangodb::aql::AstNode const* node) {
   }
 }
 
+/// @brief validate fields that start or end with ":"
+void Index::validateFieldsWithSpecialCase(VPackSlice const& fields) {
```

Comment:
Slices are lightweight and only contain a data member, which is a pointer. They can and should be passed by value.
```suggestion
void Index::validateFieldsWithSpecialCase(VPackSlice fields) {
```